### DISCUSSION TITLE
Prometheus And Kiali

### DISCUSSION THREAD

**Discussion by Vaccano** [kiali/kiali#5346] [Q&A] [answered]
I have been using istio for its ingress abilities and have really liked it.  I am now trying out Kiali.  I got it installed following the installation instructions found here:  https://kiali.io/docs/installation/installation-guide/

Once I fired up the Kiali UI I get this error on several of the pages:

```
Error Loading Graph
Cannot load the graph: Post "http://prometheus.istio-system:9090/api/v1/query": dial tcp: 
lookup prometheus.istio-system on 10.62.8.10:53: no such host
```

None of the instructions for installing Kiali said anything about prometheus.  Is it just implied that you will have installed and configured Prometheus at the url `prometheus.istio-system` on port `9000` in order to use Kiali?  Is there any other special configuration needed to use it with Kiali?  

I have not setup Prometheus for my cluster (we use Dynatrace), and it seems like a fair amount of work to do it correctly.  I am willing to do that, but I want to make sure that I am right in assuming that I have to install it in order to make use of many of Kiali's features.

**Comment by jmazzitelli** [comment]
kiali requires istio telemetry to be stored in prometheus. see docs: https://kiali.io/docs/configuration/p8s-jaeger-grafana/#prometheus-configuration

**Comment by jshaughn** [comment, marked as answer]
> None of the instructions for installing Kiali said anything about prometheus. 

It's literally line 1 in the [Installation Guide Prerequesites](https://kiali.io/docs/installation/installation-guide/prerequisites/#istio) section :-) 

"Before you install Kiali you must have already installed Istio along with its telemetry storage addon (i.e. Prometheus)."

It should be easy to install Prometheus, at least as a a demo-level addon, if you want to just take a look at Kiali.  See https://istio.io/latest/docs/setup/getting-started/#dashboard.

**Reply by Vaccano** [reply]
Wow!  I feel stupid!  I totally missed that line. And it is the very first one too!  (I think I was assuming there would be instructions on how to set it up for Istio.)

Thank you!
